tensorbay/opendataset/BDD100K/loader.py

```diff
@@ -550,39 +555,64 @@ def _load_tracking_segment(
     dataset: Dataset,
     images_directory: str,
     labels_directory: str,
-    load_label: _DATA_GETTER,
+    tracking_type: str,
 ) -> None:
     for segment_prefix in _SEGMENT_NAMES:
         image_directory = glob(os.path.join(images_directory, segment_prefix, "*"))
-        labels_directory_segment = os.path.join(labels_directory, segment_prefix)
+        segment_labels_directory = os.path.join(labels_directory, "polygons", segment_prefix)
+        original_mask_directory = os.path.join(labels_directory, "bitmasks", segment_prefix)
+        mask_directory = os.path.join(labels_directory, "single_channel_masks", segment_prefix)
+        os.makedirs(mask_directory, exist_ok=True)
+
         if segment_prefix == "test":
             generate_data: _DATA_GENERATOR = _generate_test_data
         else:
             generate_data = _generate_data
         for image_subdir in image_directory:
             segment = dataset.create_segment(f"{segment_prefix}_{os.path.basename(image_subdir)}")
-            segment.extend(generate_data(image_subdir, labels_directory_segment, load_label))
+            segment.extend(
+                generate_data(
+                    image_subdir,
+                    segment_labels_directory,
+                    original_mask_directory,
+                    mask_directory,
+                    tracking_type,
+                )
+            )
 
 
-def _generate_test_data(image_subdir: str, _: str, __: _DATA_GETTER) -> Iterable[Data]:
+def _generate_test_data(image_subdir: str, _: str, __: str, ___: str, ____: str) -> Iterable[Data]:
     yield from map(Data, glob(os.path.join(image_subdir, "*.jpg")))
 
 
 def _generate_data(
     image_subdir: str,
-    labels_directory_segment: str,
-    get_data: _DATA_GETTER,
+    segment_labels_directory: str,
+    original_mask_directory: str,
+    mask_directory: str,
+    tracking_type: str,
 ) -> Iterable[Data]:
-    label_filename = f"{os.path.basename(image_subdir)}.json"
-    with open(os.path.join(labels_directory_segment, label_filename), "r") as fp:
+    subdir_name = os.path.basename(image_subdir)
+    if tracking_type == "mots":
+        original_mask_subdir = os.path.join(original_mask_directory, subdir_name)
+        mask_subdir = os.path.join(mask_directory, subdir_name)
+        os.makedirs(mask_subdir, exist_ok=True)
+    with open(os.path.join(segment_labels_directory, f"{subdir_name}.json"), "r") as fp:
         label_contents = json.load(fp)
     for label_content in label_contents:
         label_content_name = label_content["name"]
         if "/" in label_content_name:
             label_content_name = label_content_name[len(label_content["videoName"]) + 1 :]
         image_path = os.path.join(image_subdir, label_content_name)
-
-        yield get_data(image_path, label_content)
+        yield _get_mot_data(
+            image_path, label_content
+        ) if tracking_type == "mot" else _get_mots_data(
```

Comment:
Can this `if` statement and the one in L596 be merged?

Reply:
this `if` statement is inside `for` loop, it can't be merged with the one in L596